Rephrase main_concepts/relevancy (#265)
meilisearch/documentation: guides/main_concepts/relevancy.md

# Relevancy

Search responses are sorted according to a set of consecutive rules called **ranking rules**. When a search query is made, MeiliSearch uses a [bucket sort](/guides/advanced_guides/bucket_sort.md) to rank documents. Each rule is applied to all documents that are considered equal according to the previous rule to break the tie.

> Ranking rules are **predefined rules applied to the search results** in order to improve their relevance. In order to benefit from the ranking rules and make them meet your dataset and needs, it is important to understand how each of them works and how to create new ones.

Author: Maybe replace `predefined` by `built-in` like you suggested?

For a more in-depth explanation of the algorithm and the default ranking rules, [see this issue](https://github.com/meilisearch/MeiliSearch/issues/358).

## Ranking rules

> Ranking rules are built-in rules applied in a default order to the search results to meet most standard needs. **This order can be changed in the settings**. The ranking rules are **customizable** which means **existing rules can be deleted and new ones can be added.**

Author: Much better, thx!

Each of the rules has a special use in finding the right documents for a given search query.

> The order in which the rules are set affects their importance. They are arranged from the most important rule to the least important rule.

Author: Beautiful term!

### Exactness

`Knight Moves` is displayed before `Knights of Badassdom`. `Knight` is exactly the same as the search query `Knight` whereas there is a letter of difference between `Knights` and the search query `Knight`.

:::

::::

## Order of the rules

By default, the built-in rules are executed in the following order to meet most standard needs.

```json
["typo", "words", "proximity", "attribute", "wordsPosition", "exactness"]
```

Depending on your needs, you might want to change this order of importance. To do so, you can use the [settings route](/references/ranking_rules.md#update-ranking-rules) of your index.

## Adding your rules

New rules can be added to the existing list at any time and placed anywhere in the sequence.

A custom rule allows you to create an ascending or descending sorting rule on a given attribute.

> To add your own ranking rule, you have to communicate either `asc` for ascending order or `dsc` for descending order followed by the field name in brackets.

Reviewer: Its either round brackets or parenthesis. Brackets is this `[ ]` 
```suggestion
To add your own ranking rule, you have to communicate either `asc` for ascending order or `desc` for descending order followed by the field's attribute between round brackets.
```

Author: I can't commit this suggestion.

- To apply an **ascending sorting** (results sorted by increasing value of the attribute): `asc(attribute_name)`

- To apply a **descending sorting** (results sorted by decreasing value of the attribute): `dsc(attribute_name)`

Add this rule to the existing list of ranking rules using the [settings route](/references/ranking_rules.md#update-ranking-rules).

#### Example

Let's say you have a dataset of movies. The documents contain the fields `release_date` with a timestamp as value, and `movie_ranking` an integer that represents its ranking.

> The following example will create a rule that makes older movies more relevant than older ones. A movie released in 1999 will appear before a movie released in 2020.

Reviewer: ```suggestion
The following example will create a rule that makes older movies more relevant than more recent ones. A movie released in 1999 will appear before a movie released in 2020.
```

```
asc(release_date)
```

> The following example will create a rule that makes movies with a good rank more relevant than others. Movies with a higher ranking will appear first.

Reviewer: ```suggestion
The following example will create a rule that makes movies with a good rank more relevant than movies with a lower rank. Movies with a higher ranking will appear first.
```

Author: Much better!

```

> dsc(movie_ranking)

Reviewer: ```suggestion
desc(movie_ranking)
```

```

> To add a newly created rule to the existing ranking rule using the [settings route](/references/ranking_rules.md#update-ranking-rules), you have to add the rule to the existing order array.

Reviewer: Easier to read.
```suggestion
To add a rule to the existing ranking rule, you have to add the rule to the existing ordered rules array using the [settings route](/references/ranking_rules.md#update-ranking-rules),
```

### Exactness

`Knight Moves` is displayed before `Knights of Badassdom`. `Knight` is exactly the same as the search query `Knight` whereas there is a letter of difference between `Knights` and the search query `Knight`.

:::

::::

## Order of the rules

By default, the built-in rules are executed in the following order to meet most standard needs.

```json
["typo", "words", "proximity", "attribute", "wordsPosition", "exactness"]
```

Depending on your needs, you might want to change this order of importance. To do so, you can use the [settings route](/references/ranking_rules.md#update-ranking-rules) of your index.

## Adding your rules

New rules can be added to the existing list at any time and placed anywhere in the sequence.

A custom rule allows you to create an ascending or descending sorting rule on a given attribute.

> To add your own ranking rule, you have to communicate either `asc` for ascending order or `desc` for descending order followed by the field name in brackets.

Reviewer: Its either round brackets or parenthesis. Brackets is this `[ ]` 
```suggestion
To add your own ranking rule, you have to communicate either `asc` for ascending order or `desc` for descending order followed by the field name between round brackets.
```

  "words",
  "wordsPosition",
  "exactness",
  "asc(release_date)",

>   "dsc(movie_ranking)"

Reviewer: ```suggestion
  "desc(movie_ranking)"
```

]
```

## Importance of the attributes

In a dataset, some fields are more relevant to the search than others. A `title`, for example, has a value more meaningful to a movie search than its `description` or its `director` name.

By default, the order of importance of the attributes is based on their order of appearance in the first document added. Then, each new attribute found in new documents will be added at the end of this ordered list.

If you wish to specify the order of the attributes you can either define them in the settings or set the correct order in the first document indexed.

### Changing the order of the attributes

> You may want to change the order once the documents have been ingested. This is still very possible using the searchable attributes list.

Reviewer: Could you add a link to the guide that talk about searchable attributes. Like this : [searchable attributes list](#link)
```suggestion
You may want to change the order once the documents have been ingested. This is still very possible using the searchable attributes list.
```

Author: Done!

Whenever a document is added to MeiliSearch, all new attributes found in it are automatically added to two lists:

- **The [searchable attributes list](/references/searchable_attributes.md)**: Attributes of the fields in which to search for matching query words.
- **The [displayed attributes list](/references/displayed_attributes.md)**: Attributes of the fields displayed in documents.

This searchable attributes list is **ordered**, which means the order in which the attributes appear in the list determines their relevancy. Attributes are arranged from the most important attribute to the least important attribute.

> Place the attributes in the desired order and send this updated list using the settings routes. Attributes will be re-ordered.

Reviewer: Could you add the link to setting route on "settings routes". 
```suggestion
Place the attributes in the desired order and send this updated list using the settings routes. Attributes will be re-ordered.
```

Author: Sure!
